src/player.cpp

```diff
@@ -4039,17 +4039,49 @@ void Player::addUnjustifiedDead(const Player* attacked)
 
 	sendTextMessage(MESSAGE_EVENT_ADVANCE, "Warning! The murder of " + attacked->getName() + " was not justified.");
 
-	skullTicks += g_config.getNumber(ConfigManager::FRAG_TIME);
+	Skulls_t playerSkull = getSkull();
+	if (playerSkull == SKULL_ORANGE) {
```

Comment:
As far i understand this orange skull check here does not make sense. The orange skull must be checked on the attacked player (the player which has been killed) and its a "client skull" (a skull that only appear to you at certain conditions like yellow skull, party skull, warmode skull etc) manipulated by the Creature::getSkullClient function.